actions/marketplace.ts

```diff
@@ -66,22 +106,34 @@ export function installPlugin(id: string, version: string) {
             return;
         }
 
-        await dispatch(fetchPlugins());
+        await dispatch(fetchListing());
         dispatch({
             type: ActionTypes.INSTALLING_MARKETPLACE_PLUGIN_SUCCEEDED,
             id,
         });
     };
 }
 
-// filterPlugins sets a search filter for marketplace plugins, fetching the latest data.
-export function filterPlugins(filter: string): ActionFunc {
-    return async (dispatch: DispatchFunc) => {
-        dispatch({
-            type: ActionTypes.FILTER_MARKETPLACE_PLUGINS,
-            filter,
-        });
+// installApp installed an App using a given URL via the /apps install slash command.
+//
+// On success, it also requests the current state of the plugins to reflect the newly installed plugin.
+export function installApp(id: string, url: string) {
+    return async (dispatch: DispatchFunc, getState: GetStateFunc): Promise<void> => {
+        const state = getState() as GlobalState;
+
+        const channelID = getCurrentChannelId(state);
+        const teamID = getCurrentTeamId(state);
+
+        const app = getApp(state, id);
+        if (!app) {
+            return;
+        }
+
+        const args: CommandArgs = {
+            channel_id: channelID,
+            team_id: teamID,
+        };
 
-        return dispatch(fetchPlugins());
+        dispatch(executeCommand('/apps install --force --url ' + url + ' --app-secret some-secret', args));
```

Comment:
@levb, could Ben call the InstallApp API you were working on in a wip PR directly here?